src/v/cluster/self_test/tests/self_test_request_test.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "bytes/iobuf.h"
+#include "cluster/self_test_rpc_types.h"
+
+#include <seastar/testing/thread_test_case.hh>
+
+SEASTAR_THREAD_TEST_CASE(test_make_netcheck_request) {
+    static const auto check_frags =
+      [](iobuf::const_iterator begin, const iobuf::const_iterator& end) {
+          while (begin != end) {
+              if (begin->size() != 8192) {
+                  return false;
+              }
+              begin++;
+          }
+          return true;
+      };
+
+    auto req = cluster::make_netcheck_request(model::node_id{0}, 52).get0().buf;
+    BOOST_CHECK_EQUAL(req.size_bytes(), 52);
+    BOOST_CHECK_EQUAL(std::distance(req.cbegin(), req.cend()), 1);
+    BOOST_CHECK_EQUAL(req.cbegin()->size(), 52);
```

Comment:
Maybe add a high level comment describing what this is doing? E.g. "Check that all buffers have size 8192 except the last, which fills in any remaining space." or somesuch